### PR TITLE
Deprecating WP8, WP8.1 & Windows Store 8 projects & Adding solutions for VS 2017

### PR DESCRIPTION
**Pull request by pvbchak**

### CODE REVIEW

File: src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.csproj [hunk 1]
```diff
@@ -1,117 +1,105 @@
...
-    <DebugType>pdbonly</DebugType>
-    <PlatformTarget>x86</PlatformTarget>
-    <UseVSHostingProcess>false</UseVSHostingProcess>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-  </PropertyGroup>
-  <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
-  <Import Project="..\..\packages\Microsoft.Bcl.Build.1.0.21\build\Microsoft.Bcl.Build.targets" Condition="Exists('..\..\packages\Microsoft.Bcl.Build.1.0.21\build\Microsoft.Bcl.Build.targets')" />
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (dhei) [inline src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.csproj]: Can you rename path "src/Microsoft.WindowsAzure.MobileServices.WindowsStore" to "src/Microsoft.WindowsAzure.MobileServices.UWP"?

File: src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.nuget.targets [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Condition=" '$(ExcludeRestorePackageImports)' != 'true' ">
+    <RestoreSuccess Condition=" '$(RestoreSuccess)' == '' ">True</RestoreSuccess>
+    <RestoreTool Condition=" '$(RestoreTool)' == '' ">NuGet</RestoreTool>
+    <ProjectAssetsFile Condition=" '$(ProjectAssetsFile)' == '' ">C:\git\AAPT-Zumo-SDKBuild\managed\src\Microsoft.WindowsAzure.MobileServices.WindowsStore\project.lock.json</ProjectAssetsFile>
+    <NuGetPackageRoot Condition=" '$(NuGetPackageRoot)' == '' ">$(UserProfile)\.nuget\packages\</NuGetPackageRoot>
+    <NuGetPackageFolders Condition=" '$(NuGetPackageFolders)' == '' ">C:\Users\pvbchak\.nuget\packages\</NuGetPackageFolders>
```
Comment (dhei) [inline src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.nuget.targets]: Is this nuget.targets file necessary here? Can we avoid using hardcoded file path (obviously from your local machine) here?

File: unittest/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Microsoft.WindowsAzure.Mobile.UWP.Test.nuget.targets [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Condition=" '$(ExcludeRestorePackageImports)' != 'true' ">
+    <RestoreSuccess Condition=" '$(RestoreSuccess)' == '' ">True</RestoreSuccess>
+    <RestoreTool Condition=" '$(RestoreTool)' == '' ">NuGet</RestoreTool>
+    <ProjectAssetsFile Condition=" '$(ProjectAssetsFile)' == '' ">C:\git\AAPT-Zumo-SDKBuild\managed\unittest\Microsoft.WindowsAzure.MobileServices.WindowsStore.Test\project.lock.json</ProjectAssetsFile>
+    <NuGetPackageRoot Condition=" '$(NuGetPackageRoot)' == '' ">$(UserProfile)\.nuget\packages\</NuGetPackageRoot>
+    <NuGetPackageFolders Condition=" '$(NuGetPackageFolders)' == '' ">C:\Users\pvbchak\.nuget\packages\</NuGetPackageFolders>
```
Comment (dhei) [inline unittest/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Microsoft.WindowsAzure.Mobile.UWP.Test.nuget.targets]: Same question as I asked in src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.nuget.targets.

**Comment by dhei** [comment]
LGTM

**Comment by adrianhall** [comment]
Same comments as @dhei - let's try and get absolute paths out of the projects.

File: src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.nuget.targets (hunk 2 above)
Comment (pvbchak) [inline src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.nuget.targets]: This file is auto-generated by VS & later found that these files were partially added in .gitignore. Fixing .gitignore & deleting this file.

File: unittest/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Microsoft.WindowsAzure.Mobile.UWP.Test.nuget.targets (hunk 3 above)
Comment (pvbchak) [inline unittest/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Microsoft.WindowsAzure.Mobile.UWP.Test.nuget.targets]: This file is auto-generated by VS & later found that these files were partially added in .gitignore. Fixing .gitignore & deleting this file.

File: src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.csproj (hunk 1 above)
Comment (pvbchak) [inline src/Microsoft.WindowsAzure.MobileServices.WindowsStore/Microsoft.WindowsAzure.Mobile.Ext.UWP.csproj]: Done.

**Comment by dhei** [comment]
The latest version look good to me

**Comment by adrianhall** [comment]
Likewise- go for it.

**Comment by pvbchak** [comment]
Thanks @adrianhall & @dhei for reviewing.

**Pull request merged**